server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -38,8 +38,8 @@ public abstract class DocumentParserContext {
     private static class Wrapper extends DocumentParserContext {
         private final DocumentParserContext in;
 
-        private Wrapper(DocumentParserContext in) {
-            super(in);
+        private Wrapper(ObjectMapper parent, DocumentParserContext in) {
+            super(parent, parent.dynamic == null ? in.dynamic : parent.dynamic, in);
```

Comment:
we effectively wrap the parser context for every object that we encounter while parsing a doc? Do you have a rough idea of  the effect that this may have on document parsing in case there's a high amount objects?

Reply:
These are all pretty short-lived objects, and construction is just a case of setting a bunch of object references, so I would be surprised if it had any effect at all.  We can keep an eye on indexing benchmarks to see if there's any negative effects?